File: app/src/main/java/com/example/localservicemarketplace/models/Address.java
package com.example.localservicemarketplace.models;

import java.util.UUID;

public class Address {
    private String addressId;
    private String userId;
    private String street;
    private String city;
    private String state;
    private String zipCode;
    private String country;
    private double latitude;
    private double longitude;
    private boolean isDefault;

    public Address() {
        this.addressId = UUID.randomUUID().toString();
    }

    // Getters and Setters
    public String getAddressId() { return addressId; }
    public void setAddressId(String addressId) { this.addressId = addressId; }
    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }
    public String getStreet() { return street; }
    public void setStreet(String street) { this.street = street; }
    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; }
    public String getState() { return state; }
    public void setState(String state) { this.state = state; }
    public String getZipCode() { return zipCode; }
    public void setZipCode(String zipCode) { this.zipCode = zipCode; }
    public String getCountry() { return country; }
    public void setCountry(String country) { this.country = country; }
    public double getLatitude() { return latitude; }
    public void setLatitude(double latitude) { this.latitude = latitude; }
    public double getLongitude() { return longitude; }
    public void setLongitude(double longitude) { this.longitude = longitude; }
    public boolean isDefault() { return isDefault; }
    public void setDefault(boolean aDefault) { isDefault = aDefault; }
}

File: lib/domain/entities/address.dart
import 'package:equatable/equatable.dart';

class Address extends Equatable {
  const Address({
    required this.addressId,
    required this.userId,
    required this.label,
    required this.street,
    required this.city,
    required this.state,
    required this.country,
    required this.postalCode,
    required this.latitude,
    required this.longitude,
    this.isDefault = false,
  });

  final String addressId;
  final String userId;
  final String label;
  final String street;
  final String city;
  final String state;
  final String country;
  final String postalCode;
  final double latitude;
  final double longitude;
  final bool isDefault;

  String get fullLine => '$street, $city, $state $postalCode';

  Address copyWith({
    String? addressId,
    String? userId,
    String? label,
    String? street,
    String? city,
    String? state,
    String? country,
    String? postalCode,
    double? latitude,
    double? longitude,
    bool? isDefault,
  }) {
    return Address(
      addressId: addressId ?? this.addressId,
      userId: userId ?? this.userId,
      label: label ?? this.label,
      street: street ?? this.street,
      city: city ?? this.city,
      state: state ?? this.state,
      country: country ?? this.country,
      postalCode: postalCode ?? this.postalCode,
      latitude: latitude ?? this.latitude,
      longitude: longitude ?? this.longitude,
      isDefault: isDefault ?? this.isDefault,
    );
  }

  @override
  List<Object?> get props => [
        addressId,
        userId,
        label,
        street,
        city,
        state,
        country,
        postalCode,
        latitude,
        longitude,
        isDefault,
      ];
}
